### ISSUE TITLE
Modulation Tools

### ISSUE THREAD

**Issue by qubitrenegade** [mattorp/synesthesia-ecosystem#20]
It would be really nice to have tools to be able to modulate different parameters based on time.  (also a Random/S&H is always nice)

It would be really cool if you could assign for instance, an LFO to the direction parameter and over time go from moving forward to backward, etc.

An LFO should be assignable to multiple parameters at once.

Also the ability to create multiple LFOs would be nice, so they can be assigned to different things.

Modulating LFOs with other LFOs is another fun thing to think about.

Might be an "up sell feature", trial gets 1, standard gets 8, pro gets unlimited.

**Comment by mattorp** [comment]
<https://github.com/mattorp/syn-server> can control the parameters with POST requests, and read existing syn variables with GET requests. So hopefully this can handle some of it.

I'll try to improve the documentation soon. If anything is unclear please open an issue in that repo.

**Comment by qubitrenegade** [comment]
I feel like having to install, manage, and deal with python (specifically in windows) is going to be a non-starter for a lot of people.  I have experience with python and just the thought of trying to install it on Windows again makes me... l dunno, want to run away...  :)

Also, having in-built modulation tools would allow the modulation to use the same clock source as Syn instead of trying to sync multiple clock sources together.

Thanks!

**Comment by mattorp** [comment]
Yep, python and other dev tools will surely keep many away. Sadly I don't have much time to implement more accessible toolings in the near future. Gotta love that macOS includes python by default, although only 2.7 

But maybe an installed helper tool could let web or electron apps use the functionality that pyhton handles better. I think web apps could be a nice way to let tools talk together easily, to create a proper ecosystem for external tools. Web frameworks are also incredible for developing UIs that might be difficult in other languages.

**Comment by mattorp** [comment]
Bundling may actually not be that far away: https://github.com/mattorp/syn-server/issues/8

**Comment by qubitrenegade** [comment]
Interesting...  I'd be happy to help you setup a pipeline to auto build
that as a package for you, if you'd like.

**Comment by mattorp** [comment]
Awesome, thanks! I'm a bit pressed for time for the next four weeks, so that would be amazing. Can you let me know what you need from me?

**Comment by mattorp** [comment]
@qubitrenegade https://github.com/mattorp/syn-server/issues/10

Bundling is not needed 😃

Unless you see a reason for keeping syn-server as python.

**Comment by mattorp** [comment]
Supported natively now by using setControl in the script.js
